app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationController.scala

```diff
@@ -146,16 +149,22 @@ class BusinessIdentificationController @Inject()(
     }
   }
 
-  private def validatedBusinessDetailsAndRedirect(existingSession: AgentSession)(implicit hc: HeaderCarrier): Call =
+  private def validatedBusinessDetailsAndRedirect(existingSession: AgentSession, continueUrl: Option[ContinueUrl])(
+    implicit hc: HeaderCarrier): Future[Call] =
     businessDetailsValidator.validate(existingSession.registration) match {
       case Failure(responses) if responses.contains(InvalidBusinessName) =>
         routes.BusinessIdentificationController.showBusinessNameForm()
       case Failure(responses) if responses.exists(r => r == InvalidBusinessAddress || r == DisallowedPostcode) =>
         routes.BusinessIdentificationController.showUpdateBusinessAddressForm()
       case Failure(responses) if responses.contains(InvalidEmail) =>
         routes.BusinessIdentificationController.showBusinessEmailForm()
-      case _ =>
+      case _ if continueUrl.isDefined =>
```

Comment:
I guess this fails for users who login with unclean creds ( ? ) for users with unclean creds there will be a continueUrl as far as I remember , but please double check that we are not breaking the journey ...

Reply:
its for users coming from Trusts. John informed me that only users coming from Trusts will have a continue url and we don't want them to see the task list because they haven't paid for it or some other reasons Ian gave me.